### DISCUSSION TITLE
Table improvements

### DISCUSSION THREAD

**Discussion by mjashanks** [Budibase/budibase#1913] [Ideas]
I am starting this thread as a collection of improvements to the `Table` component. 

### Change column width
https://github.com/Budibase/budibase/discussions/1821
https://github.com/Budibase/budibase/discussions/1694
https://github.com/Budibase/budibase/discussions/2749

### Selected row 
https://github.com/Budibase/budibase/discussions/1723
https://github.com/Budibase/budibase/discussions/1598
https://github.com/Budibase/budibase/discussions/3034

### Change column names
https://github.com/Budibase/budibase/discussions/1811
also raised by Sypher in discord

### Add a custom column
https://github.com/Budibase/budibase/discussions/1861

### Formatting data
https://github.com/Budibase/budibase/discussions/1597

### Colouring rows and cells
https://github.com/Budibase/budibase/discussions/3923

### Export Data
https://github.com/Budibase/budibase/discussions/1885

### Easier column picking and re-ordering 
If you have a lot of columns, it's arduous to pick columns
https://github.com/Budibase/budibase/discussions/3674

### Conditional show/hide of columns
Show/hide columns based on roles: https://github.com/Budibase/budibase/discussions/3674

Also this issue has a bunch of suggestions: https://github.com/Budibase/budibase/issues/3273

**Comment by aptkingston** [comment]
I'd propose adding a new setting component for tables similar to how navigation links are now configured, which opens a drawer:
![image](https://user-images.githubusercontent.com/9075550/124456320-739ba800-dd82-11eb-8556-b35c5d803090.png)

Here users could add, delete and reorder columns. Each column could have a text field for the pretty label, and could also have a select with some preset options for how to format the data (currency, number etc), or just a HBS field if we want to be more flexible. We could have additional settings for each column for things like "column width" or "show on mobile".

I can see a nice settings component like this working really well for the table.

**Comment by marco-santulli** [comment]
Hi,

is there any plan to add some of this functionality soon?  I think it would help a lot if we could format the content of cells and do the other things outlined here.  Alternatively it would be ok to just use repeater and fields but adding a way to use "fixed spacing because otherwise it looks bad in the page?

M

**Reply by mjashanks** [reply]
Hi @marco-santulli. I agree with you, we should aim to fit it into the roadmap somewhere. The table component is so important. I have raised this as an issue: https://github.com/Budibase/budibase/issues/3705 so that we can prioritise it alongside the roadmap.
